### PR TITLE
[v2]stop with an error immediately if a file or directory with that name already exists

### PR DESCRIPTION
**Pull request by zwwhdls**
fix https://github.com/helm/helm/issues/7182

same as https://github.com/helm/helm/pull/7187

Signed-off-by: zwwhdls <zwwhdls@hotmail.com>

### CODE REVIEW

File: cmd/helm/fetch.go [hunk 1]
```diff
@@ -166,13 +166,13 @@ func (f *fetchCmd) run() error {
...
-		if fi, err := os.Stat(ud); err != nil {
+		if _, err := os.Stat(ud); err != nil {
 			if err := os.MkdirAll(ud, 0755); err != nil {
 				return fmt.Errorf("Failed to untar (mkdir): %s", err)
 			}
 
-		} else if !fi.IsDir() {
-			return fmt.Errorf("Failed to untar: %s is not a directory", ud)
+		} else {
+			return fmt.Errorf("ailed to untar: a file or directory with the name %s already exists", ud)
```
Comment (karuppiah7890) [inline cmd/helm/fetch.go]: a small typo `failed`

**Review by karuppiah7890 — CHANGES_REQUESTED** [review]

File: cmd/helm/fetch.go (hunk 1 above)
Comment (zwwhdls) [reply cmd/helm/fetch.go]: 🤣fixed.

**Review by technosophos — APPROVED** [review]
This might be seen as "breaking backward compatibility", but I would argue that it is a bug fix. We shouldn't let the user partially overwrite a chart because it will lead to really hard to debug problems.

Fix LGTM

**Comment by karuppiah7890** [comment]
I just wanted to add this - could we check for the error message in the test? To be sure that it's the expected error message. And also have separate tests for file and directory, both with same name as chat being extracted/untarred

**Review by hickeyma — CHANGES_REQUESTED** [review]
We should check on the clashing directory in the message as per review for Helm 3 https://github.com/helm/helm/pull/7187#pullrequestreview-330454226. Lets hold on this PR until v3 is merged first.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks for porting this @zwwhdls

**Pull request merged**